tests/src/test/java/alluxio/client/cli/JournalToolTest.java

```
@@ -112,23 +196,114 @@ public void dumpHeapCheckpoint() throws Exception {
+      assertNonemptyFileExists(PathUtils.concatPath(fsMasterCheckpointsDir, "INODE_TREE", subPath));
+    }
+  }
+
+  @Test
+  public void dumpRocksCheckpoint() throws Throwable {
+    initializeCluster(new HashMap<PropertyKey, String>() {
+      {
+        put(PropertyKey.MASTER_METASTORE, "ROCKS");
+      }
+    });
+
+    checkpointUfsJournal();
     JournalTool.main(new String[] {"-outputDir", mDumpDir.getAbsolutePath()});
     String checkpointDir = findCheckpointDir();
     assertNonemptyDirExists(
         PathUtils.concatPath(checkpointDir, "INODE_TREE", "CACHING_INODE_STORE"));
   }
 
-  private void checkpoint() throws Exception {
+  private void checkpointUfsJournal() throws Exception {
     // Perform operations to generate a checkpoint.
     for (int i = 0; i < CHECKPOINT_SIZE * 2; i++) {
       mFs.createFile(new AlluxioURI("/" + i)).close();
     }
     IntegrationTestUtils.waitForUfsJournalCheckpoint(Constants.FILE_SYSTEM_MASTER_NAME);
   }
 
+  private void checkpointEmbeddedJournal() throws Throwable {
+    int leaderIdx = mMultiProcessCluster.getPrimaryMasterIndex(GET_MASTER_INDEX_TIMEOUT_MS);
+    int followerIdx = leaderIdx ^ 1; // Assumes 2 masters.
+    String followerJournalFolder =
+        mMultiProcessCluster.getJournalDir() + Integer.toString(followerIdx);
+
+    // Get current snapshot before issuing operations.
+    long curentSnapshotIdx = getCurrentCopyCatSnapshotIndex(followerJournalFolder);
+
+    // Perform operations to generate a checkpoint.
+    for (int i = 0; i < CHECKPOINT_SIZE * 2; i++) {
+      mFs.createFile(new AlluxioURI("/" + i)).close();
+    }
+
+    // Wait until current snapshot index changes.
+    CommonUtils.waitFor("copycat checkpoint to be written", () -> {
+      try {
+        long latestSnapshotIdx = getCurrentCopyCatSnapshotIndex(followerJournalFolder);
+        return latestSnapshotIdx != curentSnapshotIdx;
+      } catch (Throwable err) {
+        return false;
```

Are the errors expected? If not, shall we throw the error to bail out early?

---

There could be errors.
Throwing it now.